gRPC/Client Examples/StimTherapyApp/MainWindow.xaml.cs

```diff
@@ -153,66 +155,96 @@ private void MainWindow_Loaded(object sender, RoutedEventArgs e)
                     // disable buttons
                     btn_beta.IsEnabled = false; // beta stim button; have to use method invoker
                     btn_open.IsEnabled = false; // open loop stim button
-                    btn_diagnostic.IsEnabled = false; // diagnostics button
                     btn_stop.IsEnabled = false; // stop stim button
 
-                    // temporary- hide diagnostic buttons
-                    btn_diagnostic.Visibility = Visibility.Hidden;
                 }));
 
             // Start update timer
             neuroChartUpdateTimer = new System.Timers.Timer(200);
             neuroChartUpdateTimer.Elapsed += neuroChartUpdateTimer_Elapsed;
             neuroChartUpdateTimer.Start();
-        }
 
-        private void neuroChartUpdateTimer_Elapsed(object sender, ElapsedEventArgs e)
+            connectionUpdateTimer = new System.Timers.Timer(200);
+            connectionUpdateTimer.Elapsed += connectionUpdateTimer_Elapsed;
+            connectionUpdateTimer.Start();
+        }
+        
+        private void connectionUpdateTimer_Elapsed(object sender, ElapsedEventArgs e)
         {
-            // grab latest data
-            List<double>[] neuroData = aBICManager.getData();
-
-            // look for the selected items in the listbox
-            List<int> selectedChannels = new List<int>();
-            string chanString = "";
-            int chanVal;
-            bool valConvert = false;
+            List<string> connectionInfo = aBICManager.getConnectionInfo();
 
-            // get a list of selected channels
-            var selected = from item in channelList
-                           where item.IsSelected == true
-                           select item.Name.ToString();
-
-            // convert from string to int type
-            foreach (String item in selected)
+            // When connectionInfo is populated with information about disconnection
+            if (connectionInfo.Any())
             {
-                valConvert = Int32.TryParse(item, out chanVal);
-                if (valConvert)
+                ThreadPool.QueueUserWorkItem(a =>
                 {
-                    selectedChannels.Add(chanVal);
-                }
-                else
-                {
-                    selectedChannels.Add(33);
-                }
+                    Application.Current.Dispatcher.Invoke(new Action(() =>
+                    {
+                        // Notify user about disconnection event
+                        OutputConsole.Inlines.Add("Disconnection at " + connectionInfo[0] + " connection. ");
+                        OutputConsole.Inlines.Add("Check connections then use the 'Reconnect' button to reestablish connection!\n");
+                        neuroChartUpdateTimer.Stop();
```

Comment:
with the connectState check, is it needed to stop this? it's not a big problem if it just continues right?

Reply:
true- will let timer for neural streaming continue